srv/salt/ceph/smoketests/openstack/init.sls

```diff
@@ -0,0 +1,88 @@
+{% set master = salt['master.minion']() %}
+
+# NOTE: I'm unsure exactly how to structure this; possibly it would be
+# better to move the inital cleanup and subsequent keyring and pool tests
+# to SLS files in srv/ceph/tests/openstack/, and make this orchestration
+# apply those states to the master, rather than having all these
+# salt.function blocks, which don't really seem to give enough information
+# back on failure?
+
+# First we get rid of anything (users, pools) that may have been created by
+# a previous test run, so we know we're starting clean.
+
+{% for user in ['cinder', 'cinder-backup', 'glance'] %}
+remove client.smoketest-{{ user }}:
+  salt.function:
+    - name: cmd.run
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - arg:
+      - "ceph auth rm client.smoketest-{{ user }} 2>/dev/null || :"
+{% endfor %}
+
+enable pool deletion:
+  salt.function:
+    - name: cmd.run
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - arg:
+      - "ceph tell mon.* injectargs --mon-allow-pool-delete=true"
+
+{% for pool in ['images', 'volumes', 'backups', 'vms'] %}
+remove smoketest-{{ pool }}:
+  salt.function:
+    - name: cmd.run
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - arg:
+      - "ceph osd pool rm smoketest-{{ pool }} smoketest-{{ pool }} --yes-i-really-really-mean-it 2>/dev/null || :"
+{% endfor %}
+
+disable pool deletion:
+  salt.function:
+    - name: cmd.run
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - arg:
+      - "ceph tell mon.* injectargs --mon-allow-pool-delete=false"
+
+# Now, apply the openstack state, to create the relevant pools and users
+
+apply ceph.openstack:
+  salt.state:
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - sls: ceph.openstack
+    - pillar: {'openstack_prefix': 'smoketest' }
+
+# Finally, verify the pools and users exist
+
+{% for user in ['cinder', 'cinder-backup', 'glance'] %}
+verify client.smoketest-{{ user }} exists:
+  salt.function:
+    - name: cmd.run
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - arg:
+      - "ceph auth get client.smoketest-{{ user }} >/dev/null"
+{% endfor %}
+
+{% for pool in ['images', 'volumes', 'backups', 'vms'] %}
+verify smoketest-{{ pool }} exists:
+  salt.function:
+    - name: cmd.run
+    - tgt: {{ master }}
+    - tgt_type: compound
+    - arg:
+      - "rados lspools | grep '^smoketest-{{ pool }}$' >/dev/null"
+{% endfor %}
+
+# TODO:
+# - Verify the various users can access the appropriate pools and check
+#   ceph status, but can't do anything else.
+# - Delete the just-created pools and users, to leave the cluster somewhat
+#   clean-ish?
+# - Add tests for the openstack.integrate runner itself, to make sure it
+#   returns correct data.
+# - Smoketests without prefix?  This would be destructive of any existing
```

Comment:
That's what I was about to write. Adding a prefix takes a considerable ammount of different routes across the codebase. It might be wise to add a smoketest for the 'prefix route' aswell.